lib/checker/checkercomponent.cpp

```diff
@@ -162,13 +163,25 @@ void CheckerComponent::CheckThreadProc()
 
 			TimePeriod::Ptr tp = checkable->GetCheckPeriod();
 
-			if (tp && !tp->IsInside(Utility::GetTime())) {
-				Log(LogNotice, "CheckerComponent")
-					<< "Skipping check for object '" << checkable->GetName()
-					<< "': not in check period '" << tp->GetName() << "'";
+			if (tp) {
+				auto ts (Utility::GetTime());
 
-				check = false;
-				notifyNextCheck = true;
+				if (!tp->IsInside(ts)) {
+					nextCheck = tp->FindNextTransition(ts);
+
+					Log msg (LogNotice, "CheckerComponent");
+
+					msg << "Skipping check for object '" << checkable->GetName()
+						<< "', as not in check period '" << tp->GetName() << "'";
+
+					if (nextCheck > 0) {
+						nextCheck += 1;
```

Comment:
Why is this necessary? If `FindNextTransition()` returns a valid timestamp then passing that ts to `IsInside()` should always evaluate to true.
https://github.com/Icinga/icinga2/blob/d230e87cf2167aa8f74a4d1af297a9ed82972109/lib/icinga/timeperiod.cpp#L294

Reply:
Idk, I've just addressed https://github.com/Icinga/icinga2/pull/10070#discussion_r1926691267

Reply:
Alvar just suggested to use 1 second instead of the value you previously used `0.01`, and I'm telling you, unless you've a specific reason, that this is completely unnecessary and that you don't have to modify the timestamp returned by `FindNextTransition()`.